designs/2024-multithread-linting/README.md

```diff
@@ -508,6 +508,45 @@ This strategy comes with some unobvious implications that consumers should be aw
 * Errors thrown by `lintFiles()` could be of a different type than the errors thrown when running in single-thread mode, and they may leak unserializable properties.
 * Because of race conditions, subsequent calls to `lintFiles()` could produce different errors for identical input.
 
+### Warning Management
```

Comment:
I'm wondering if it would be helpful to extract warning management into a separate object rather than using `process.emitWarning()` directly. We could then have `Linter` use a default warning object that stubs out the relevant method(s) rather than emitting to console. We could then pass the "live" version to `Linter` from `ESLint`. Maybe this could simplify the thread behavior?

Reply:
This is a good idea, I've updated the RFC. The names for the new option and methods are just placeholders, so feel free to suggest better names.